test/tests/subscription/PriceUtilsTest.ts

```diff
@@ -133,6 +139,23 @@ o.spec("price utils getSubscriptionPrice", async () => {
 	})
 })
 
+o.spec("PaymentInterval", function() {
+
+	o("asPaymentInterval correct values", function() {
+		o(asPaymentInterval(1)).equals(PaymentInterval.Monthly)
+		o(asPaymentInterval(12)).equals(PaymentInterval.Yearly)
+		o(asPaymentInterval("1")).equals(PaymentInterval.Monthly)
+		o(asPaymentInterval("12")).equals(PaymentInterval.Yearly)
+	})
+
+	o("asPaymentInterval rejects invalid values", function() {
+		o(() => asPaymentInterval(0)).throws(ProgrammingError)
+		o(() => asPaymentInterval("")).throws(ProgrammingError)
+		o(() => asPaymentInterval(null as any)).throws(ProgrammingError)
```

Comment:
idk if it's necessary, this is not ruby to check for type errors

Reply:
at least the check for 0 is not entirely pointless because that's the default value of the paymentInterval property on the entities that care about these things. we want an error in case we forget to set it when creating entities.